src/org/omegat/core/data/ImportFromAutoTMX.java

```diff
@@ -93,48 +93,48 @@ void process(ExternalTMX tmx, boolean isEnforcedTMX) {
                     } else if (!existTranslation.isTranslated()) {
                         // default translation not exist - use from auto tmx
-                        setTranslation(ste, e, isDefaultTranslation, TMXEntry.ExternalLinked.xAUTO);
+                        setTranslation(ste, e, isDefaultTranslation, ProjectTMXEntry.ExternalLinked.xAUTO);
                     }
                 } else { // TMXEntry with x-ids
                     if (!existTranslation.isTranslated() || existTranslation.defaultTranslation) {
                         // need to update if id in xICE or x100PC
                         if (hasICE) {
-                            setTranslation(ste, e, false, TMXEntry.ExternalLinked.xICE);
+                            setTranslation(ste, e, false, ProjectTMXEntry.ExternalLinked.xICE);
                         } else if (has100PC) {
-                            setTranslation(ste, e, false, TMXEntry.ExternalLinked.x100PC);
+                            setTranslation(ste, e, false, ProjectTMXEntry.ExternalLinked.x100PC);
                         }
-                    } else if (existTranslation.linked == TMXEntry.ExternalLinked.xICE
-                            || existTranslation.linked == TMXEntry.ExternalLinked.x100PC) {
+                    } else if (existTranslation.linked == ProjectTMXEntry.ExternalLinked.xICE
+                            || existTranslation.linked == ProjectTMXEntry.ExternalLinked.x100PC) {
                         // already contains x-ice
                         if (hasICE
-                                && !Objects.equals(existTranslation.translation, e.translation)) {
-                            setTranslation(ste, e, false, TMXEntry.ExternalLinked.xICE);
+                                && !Objects.equals(existTranslation.getTranslationText(), e.getTranslationText())) {
+                            setTranslation(ste, e, false, ProjectTMXEntry.ExternalLinked.xICE);
                         }
-                    } else if (existTranslation.linked == TMXEntry.ExternalLinked.x100PC) {
+                    } else if (existTranslation.linked == ProjectTMXEntry.ExternalLinked.x100PC) {
                         // already contains x-100pc
                         if (has100PC
-                                && !Objects.equals(existTranslation.translation, e.translation)) {
-                            setTranslation(ste, e, false, TMXEntry.ExternalLinked.x100PC);
+                                && !Objects.equals(existTranslation.getTranslationText(), e.getTranslationText())) {
+                            setTranslation(ste, e, false, ProjectTMXEntry.ExternalLinked.x100PC);
                         }
                     }
                 }
             }
         }
     }
 
-    private boolean isAltTranslation(PrepareTMXEntry entry) {
-        if (entry.otherProperties == null) {
+    private boolean isAltTranslation(ITMXEntry entry) {
+        if (entry.getProperties() == null) {
```

Comment:
we should introduce `boolean hasProperties()` method then we can encapsulate properties field in class.

Reply:
Agree, change pushed today